### PR TITLE
New Create2  Guide

### PR DESCRIPTION
**Pull request by buddies2705**
Guide to help people create upgradable contracts using zos create2

### CODE REVIEW

File: packages/docs/docs/docs/guide_create2_2.4.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+---
+id: create2
+title: Upgradable Create2 Contracts with zOS
+---
+
+
+
+Create2 opcode gives us the ability to calculate smart contracts addresses without actually deploying them on Ethereum blockchain. This opens up a lot of possibilities to improve [user onboarding and scalability](https://blog.zeppelinos.org/getting-the-most-out-of-create2/).
```
Comment (spalladino) [inline packages/docs/docs/docs/guide_create2_2.4.md]:
```suggestion
The CREATE2 opcode gives us the ability to calculate smart contracts addresses without actually deploying them on the Ethereum blockchain. This opens up a lot of possibilities to improve [user onboarding and scalability](https://blog.zeppelinos.org/getting-the-most-out-of-create2/).
```

We'll have a grammar editor go through the text later, but please keep an eye out for these kind of errors!

File: packages/docs/docs/docs/guide_create2_2.4.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+In this guide, we will create an upgradableWallet contract with a transfer method, then we will reserve an address using Create2 without actually deploying Wallet contract and send some ethers to this address. At last, we will actually deploy our Wallet contract and execute transfer method to transfer contract funds to another account.
```
Comment (spalladino) [inline packages/docs/docs/docs/guide_create2_2.4.md]:
```suggestion
In this guide, we will create an `upgradableWallet` contract with a `transfer` method, then we will reserve an address using Create2 without actually deploying Wallet contract and send some ethers to this address. At last, we will actually deploy our Wallet contract and execute transfer method to transfer contract funds to another account.
```

Let's use inline code where appropriate

File: packages/docs/docs/docs/guide_create2_2.4.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+## Deploying Contract using Create2
+
+Before deploying, we need to add our contract usingzos add, this will add our contract to ZeppelinOS project, so it can be deployed using zos push afterward.
+
+```console 
+zos add Wallet 
+```
+
+Now we will generate contract address using zos create2, we will use 12345 as salt, you can choose any random number and--query option tells zOS that we want to compute address, not the actual deployment of the contract. We will also pass network information using -n development , you can find this configuration innetworks.js file.
```
Comment (spalladino) [inline packages/docs/docs/docs/guide_create2_2.4.md]:
```suggestion
Now we will generate contract address using zos create2, we will use 12345 as salt, you can choose any random number and --query option tells zOS that we want to compute address, not the actual deployment of the contract. We will also pass network information using -n development, you can find this configuration in networks.js file.
```

File: packages/docs/docs/docs/guide_create2_2.4.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+
+
+Create2 opcode gives us the ability to calculate smart contracts addresses without actually deploying them on Ethereum blockchain. This opens up a lot of possibilities to improve [user onboarding and scalability](https://blog.zeppelinos.org/getting-the-most-out-of-create2/).
+
+In this guide, we will create an upgradableWallet contract with a transfer method, then we will reserve an address using Create2 without actually deploying Wallet contract and send some ethers to this address. At last, we will actually deploy our Wallet contract and execute transfer method to transfer contract funds to another account.
+
+## Ethereum Accounts
+
+Before diving into Create2, let’s understand a few basic things about Ethereum contract accounts. A contract account
```
Comment (spalladino) [inline packages/docs/docs/docs/guide_create2_2.4.md]: Re-reading this section, it doesn't seem to be relevant for anything that comes after - only that the CREATE opcode was used for creating a contract. I'd remove it altogether.

File: packages/docs/docs/docs/guide_create2_2.4.md (hunk 2 above)
Comment (buddies2705) [reply packages/docs/docs/docs/guide_create2_2.4.md]: missed it...while converting it from Medium

File: packages/docs/docs/docs/guide_create2.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+title: Using CREATE2 for deploying precomputed addresses
+---
+
+
+
+The CREATE2 opcode gives us the ability to calculate smart contracts addresses without actually deploying them on the Ethereum blockchain. This opens up a lot of possibilities to improve [user onboarding and scalability](https://blog.zeppelinos.org/getting-the-most-out-of-create2/).
+
+In this guide, we will create an upgradable `Wallet` contract with a `transfer` method, then we will reserve an address using CREATE2 without actually deploying `Wallet` contract and send some `ethers` to this address. At last, we will actually deploy our `Wallet` contract and execute `transfer` method to transfer contract funds to another account.
+
+## Account creation using CREATE opcode
```
Comment (spalladino) [inline packages/docs/docs/docs/guide_create2.md]:
```suggestion
## Contract creation using CREATE opcode
```

**Comment by spalladino** [comment]
I think we're good to go! Left one suggestion only, and then we can merge.

**Comment by spalladino** [comment]
@buddies2705 looks good, but why did you close the PR? I think it was good to merge, right?

**Comment by buddies2705** [comment]
How to reopen this?

**Comment by spalladino** [comment]
I'll reopen and merge, no worries

**Pull request merged**